### PR TITLE
[SM-771] Add new endpoint for bulk enabling users for Secrets Manager

### PR DESCRIPTION
**Pull request by Thomas-Avery**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The purpose of this PR is to provide a new endpoint for bulk enabling users for secrets manager.

## Code changes

* **src/Api/Controllers/OrganizationUsersController.cs:**
Adding in new endpoint for bulk enabling users for secrets manager.

* **src/Core/OrganizationFeatures/OrganizationServiceCollectionExtensions.cs:**
Adding `EnableAccessSecretsManagerCommand` into dependency injection.
Since this is a command for the organization user controller, I thought this was the appropriate place for DI.

* **src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs:**
Command for bulk enabling users for secrets manager.

* **src/Core/SecretsManager/Commands/EnableAccessSecretsManager/Interfaces/IEnableAccessSecretsManagerCommand.cs:**
Interface for the command.

* **test/Core.Test/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommandTests.cs:**
Unit tests for the new command.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (eliykat) [inline src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs:17]: Please use named tuples, especially because the `string` property is not very descriptive (I had to look it up):

```suggestion
    public async Task<List<(OrganizationUser orgUser, string error)>> EnableUsersAsync(Guid organizationId,
```

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (eliykat) [inline src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs:28]: This is where we would usually set the `string error` property for that orgUser, e.g. "User already has access to Secrets Manager", and include it in the return value.

Then that error message displays in the bulk actions dialog on the front-end, next to that user. (You can see this in other bulk operations today, e.g. bulk confirm users who have already been confirmed.)

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (eliykat) [inline src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs:21]: I'm not 100% sure on this, but I wonder if this validation should be done outside the command. Do we tell the command, "enable SM for these orgUsers for this organization"? Or should we just tell it, "enable SM for these orgUsers"? Specifying the organization seems redundant; should the command just trust what's passed into it?

I prefer the latter, but I don't know if there's a right or wrong answer, so it's your call. Just thought I'd mention it for discussion.

**Review by eliykat — CHANGES_REQUESTED** [review]
This is only a high level review on a couple of things that jumped out at me, so please get a second review from SM Team as well. Thanks!

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (Thomas-Avery) [reply src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs]: Thanks for taking a look! 

Resolved in https://github.com/bitwarden/server/pull/3020/commits/015c68aa31ccdeb0b246f7d3a903eb1312174b3d

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (Thomas-Avery) [reply src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs]: Resolved in https://github.com/bitwarden/server/pull/3020/commits/015c68aa31ccdeb0b246f7d3a903eb1312174b3d

File: src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs
Comment (Thomas-Avery) [reply src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs]: I think it makes sense to take that out of the command. It really is a query concern when thinking about the CQRS pattern. 

Resolved in https://github.com/bitwarden/server/pull/3020/commits/015c68aa31ccdeb0b246f7d3a903eb1312174b3d

File: src/Api/Controllers/OrganizationUsersController.cs
Comment (coltonhurst) [inline src/Api/Controllers/OrganizationUsersController.cs:473]: I would have expected this to be a 404 not found. Is there anything specific we are trying to accomplish by using `BadRequestException`?

**Review by coltonhurst — COMMENTED** [review]
Looking good, only one comment.

File: src/Api/Controllers/OrganizationUsersController.cs
Comment (Thomas-Avery) [reply src/Api/Controllers/OrganizationUsersController.cs:473]: I was going with what was pre-established by the other bulk requests. 

Specifically, the bulk delete.

https://github.com/bitwarden/server/blob/ca7ced4e435823edc677efcced06f5689a2e8754/src/Core/Services/Implementations/OrganizationService.cs#L1640-L1647

Comment (coltonhurst) [reply src/Api/Controllers/OrganizationUsersController.cs:473]: Makes sense, thanks for pointing that out. As usual we can leave it for consistency and address the responses in a different PR if we want.

File: test/Core.Test/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommandTests.cs
Comment (eliykat) [inline test/Core.Test/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommandTests.cs:18]: `data` is not very descriptive, I suggest keeping the same name as the parameter in the method under test. i.e. `organizationUsers`. Then it's immediately obvious what arguments you're arranging and asserting.

**Review by eliykat — APPROVED** [review]
Looks good to me, just 1 minor comment about the tests. Thanks for handling!

**Review by coltonhurst — APPROVED** [review]

**Pull request merged**
